### PR TITLE
Updated help messages for install.sh and rmake.py

### PR DESCRIPTION
**Pull request by rkamd**
resolves # [LWPMLSE-162](https://ontrack-internal.amd.com/browse/LWPMLSE-162)

Summary of proposed changes:
- Updated help messages in install.sh
- Updated help messages in rmake.py

### CODE REVIEW

**Review by TorreZuk — COMMENTED** [review]
Some suggestions to consider ,  can see what others say.

File: install.sh [hunk 1]
```diff
@@ -8,28 +8,65 @@
...
+    $0 (build hipblas and put library files at ./build/hipblas-install)
+    $0 <options> (modify default behavior according to the following flags)
+
+  Options:
+    -h | --help                   Print this help message.
+
+    -i | --install                Pass this flag to generate and install library package after build.
+
+    -d | --dependencies           Pass this flag to also build and install external dependencies.
+                                  Dependecies are to be installed in /usr/local. This should be done only once.
```
[the remark this answers is not in the document]
Comment (daineAMD) [reply install.sh:25]: Yes, see https://github.com/ROCmSoftwarePlatform/hipBLAS/blob/develop/install.sh#L441

[the remark this answers is not in the document]
Comment (rkamd) [reply install.sh:25]: Fixed the typo

File: install.sh [hunk 2]
```diff
@@ -8,28 +8,65 @@
...
+  hipBLAS library build & installation helper script.
+
+  Usage:
+    $0 (build hipblas and put library files at ./build/hipblas-install)
+    $0 <options> (modify default behavior according to the following flags)
+
+  Options:
+    -h | --help                   Print this help message.
+
+    -i | --install                Pass this flag to generate and install library package after build.
```
[the remark this answers is not in the document]
Comment (rkamd) [reply install.sh:22]: I agree but my preferrence is to keep it as "Pass this flag", for the following reasons
- In hipcc "Pass" phrase is used for options taking arguments such as "Pass <arg>". Here, the options do no take any args, so I think it is fine to explicity mention to the users that these are flags. 
- It matches the rocSOLVER help message
@amcamd / @daineAMD,
Let me know your preference, I will update rocBLAS install script accordingly.

File: install.sh [hunk 3]
```diff
@@ -8,28 +8,65 @@
...
+    -r | --relocatable            Pass this to add RUNPATH(based on ROCM_RPATH) and remove ldconf entry.
+
+    -k | --relwithdebinfo         Pass this flag to build in release debug mode (equivalent to set CMAKE_BUILD_TYPE=RelWithDebInfo).
+                                  (Default build type is Release)
+
+    --rocblas-path <blasdir>      Specify path to an existing rocBLAS install directory.
+                                  (e.g. /src/rocBLAS/build/release/rocblas-install)
+
+    -b | --rocblas                Pass this flag to specify rocblas version.
+
```
Comment (amcamd) [inline install.sh:48]: should this be 
-b | --rocblas <version>

File: install.sh [hunk 4]
```diff
@@ -8,28 +8,65 @@
...
+    --no-hip-clang                Pass this flag to build library without using hip-clang compiler.
+
+    --cuda | --use-cuda           Pass this flag to build library for cuda backend.
+
+    -p | --cmakepp                Pass this flag for to add CMAKE_PREFIX_PATH
+
+    --custom-target <target>      Pass this flag to link the library against custom target (eg. host, device).
+
+    -v | --rocm-dev <version>     Pass this flag to set specific rocm-dev version.
+
```
Comment (amcamd) [inline install.sh:65]: I think this should be:
Specify rocm-dev version (eg. 4.5.0)

File: install.sh [hunk 5]
```diff
@@ -8,28 +8,65 @@
...
+    --rocblas-path <blasdir>      Specify path to an existing rocBLAS install directory.
+                                  (e.g. /src/rocBLAS/build/release/rocblas-install)
+
+    -b | --rocblas                Pass this flag to specify rocblas version.
+
+    --rocsolver-path <solverdir>  Specify path to an existing rocSOLVER install directory.
+                                  (e.g. /src/rocSOLVER/build/release/rocsolver-install)
+
+    --compiler </compier/path>    Specify path to host compiler.
```
[the remark this answers is not in the document]
Comment (amcamd) [reply install.sh:52]: can you add an (eg. /opt/rocm/bin)  you will need to test that the example works

**Comment by rkamd** [comment]
> I think this should be:
>Specify rocm-dev version (eg. 4.5.0) 

Agree

File: install.sh [hunk 6]
```diff
@@ -8,28 +8,65 @@
...
+    --hip-clang                   Pass this flag to build using the hip-clang compiler.
+
+    --no-hip-clang                Pass this flag to build library without using hip-clang compiler.
+
+    --cuda | --use-cuda           Pass this flag to build library for cuda backend.
+
+    -p | --cmakepp                Pass this flag for to add CMAKE_PREFIX_PATH
+
+    --custom-target <target>      Pass this flag to link the library against custom target (eg. host, device).
+
```
Comment (amcamd) [inline install.sh:63]: I think this should be:
Specify custom target to link the library against (eg. host, device)  

You will need to check that the examples work

File: install.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (amcamd) [reply install.sh:22]: My preference would be to drop the "Pass this flag" when it is only a flag, and use "Specify" when it is a flag + a string, so:
- Print this help message
- Generate ans install library package after build
- Build and install external dependencies. Dependencies are to be installed in /usr/local. This should be done only once.
- ...
- Specify path to an existing rocBLAS install directory (eg. /home/rocBLAS/build/release/rocblas-install)
- Specify rocBLAS version (eg. 2.42.0)
- Specify path to an existing rocSOLVER install directory (eg. /home/rocSOLVER/build/release/rocsolver-install)

File: install.sh (hunk 3 above)
Comment (rkamd) [reply install.sh:48]: Yes, I will fix it

**Comment by rkamd** [comment]
Shall I sort the options in alphabetical order? @amcamd,@torrezuk,@daineAMD

**Comment by rkamd** [comment]
In the last commit, I have addressed the Andrew's review comments and also re-arranged the options in sorted order. In the install.sh script I replaced '|' symbol to a comma which matches the format used by linux commands.

**Review by amcamd — APPROVED** [review]
This is a nice improvement, should make install.sh more user friendly.

**Comment by amcamd** [comment]
There is more information on --custom-target here: https://github.com/ROCm-Developer-Tools/HIP/issues/2158

There is more information on --rocm-dev here: https://github.com/ROCmSoftwarePlatform/rocBLAS/pull/1040

Both of the above options are for expert users, the first specifies the target device for the compiler. The second specifies an install base for the rocm dev-package

**Pull request merged**